update-asdf-and-dockerfile/action.yml

```diff
@@ -116,7 +116,7 @@ runs:
       shell: bash
       run: |
         test -f pyproject.toml && \
-        sed -i "s/^python =.*/python = \"^${{ env.LATEST_VERSION }}\"/" pyproject.toml && \
+        sed -i "s/^python =.*/python = \"${{ env.LATEST_VERSION }}\"/" pyproject.toml && \
```

Comment:
do we need `~` here?

Reply:
If we want a fixed version we would not need the `~` : https://python-poetry.org/docs/cli/#add

Reply:
Now I know it, thank you